### PR TITLE
Add BUILD_OS to Telemetry

### PR DESCRIPTION
**Pull request by AlexFenlon**
### Proposed changes

This collects the base Build_OS whenever an image is build and adds it to telemetry as shown below.

snippet of the log for when debian image with nginx plus is deployed:
`BuildOS:debian-plus}}`

snippet of the log for when alpine image with nginx oss is deployed:
`BuildOS:alpine}}`

snippet of the log for when debian image with nginx oss is deployed:
`BuildOS:debian}}`

Resolves: #5954 

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -20,6 +20,9 @@ FROM golang:1.22-alpine@sha256:8c9183f715b0b4eca05b8b3dbf59766aaedb41ec07477b132
 
 ############################################# Base image for Alpine #############################################
 FROM nginx:1.27.0-alpine@sha256:a45ee5d042aaa9e81e013f97ae40c3dda26fbe98f22b6251acdf28e579560d55 AS alpine
+ARG BUILD_OS=alpine
+
+ENV BUILD_OS=${BUILD_OS}
```
Comment (pdabelf5) [inline build/Dockerfile:25]: I'm not sure these are needed, just setting the `ENV` in `common` might be enough

File: internal/telemetry/collector.go [hunk 2]
```diff
@@ -334,5 +341,6 @@ func (c *Collector) BuildReport(ctx context.Context) (Report, error) {
 		AppProtectVersion:    appProtectVersion,
 		IsPlus:               isPlus,
 		InstallationFlags:    installationFlags,
+		BuildOS:              buildOS,
```
Comment (pdabelf5) [inline internal/telemetry/collector.go:344]:
```suggestion
		BuildOS:              c.BuildOS(),
```

File: internal/telemetry/collector_test.go [hunk 3]
```diff
@@ -1129,6 +1129,60 @@ func TestCollectInstallationFlags(t *testing.T) {
...
+			c, err := telemetry.NewCollector(cfg, telemetry.WithExporter(exp))
+			if err != nil {
+				t.Fatal(err)
+			}
+			c.Collect(context.Background())
+
+			buildOS := c.BuildOS()
+
+			if tc.wantOS != buildOS {
+				t.Errorf("want: %s, got: %s", tc.wantOS, buildOS)
```
Comment (pdabelf5) [inline internal/telemetry/collector_test.go:1180]:
```suggestion
				t.Errorf("want: %s, got: %s", tc.wantOS, c.BuildOS())
```

File: cmd/nginx-ingress/main.go [hunk 4]
```diff
@@ -64,6 +64,9 @@ func main() {
 	parseFlags()
 	parsedFlags := os.Args[1:]
 
+	buildOS := os.Getenv("BUILD_OS")
```
Comment (jjngx) [inline cmd/nginx-ingress/main.go:67]: Is it the intention to set the var `buildOS` to an empty string if the `BUILD_OS` env var is not set?

File: internal/telemetry/collector_test.go (hunk 3 above)
Comment (pdabelf5) [reply internal/telemetry/collector_test.go:1180]: nevermind, `buildOS` is used more than once

File: cmd/nginx-ingress/main.go (hunk 4 above)
Comment (AlexFenlon) [reply cmd/nginx-ingress/main.go:67]: I am not sure, if it is not, what would the default be?

File: internal/telemetry/collector_test.go [hunk 5]
```diff
@@ -1129,6 +1129,60 @@ func TestCollectInstallationFlags(t *testing.T) {
...
 
+func TestCollectBuildOS(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name    string
+		buildOS string
+		wantOS  string
+	}{
+		{
```
Comment (jjngx) [inline internal/telemetry/collector_test.go:1140]: see previous question, what `wantOS` do we want if the `name` is an empty string?

File: cmd/nginx-ingress/main.go (hunk 4 above)
Comment (jjngx) [reply cmd/nginx-ingress/main.go:67]: https://pkg.go.dev/os#Getenv

Comment (AlexFenlon) [reply cmd/nginx-ingress/main.go:67]: It is okay if we report either null or empty string, these are both fine.

File: cmd/nginx-ingress/main.go [hunk 6]
```diff
@@ -64,6 +64,9 @@ func main() {
...
+	fmt.Println("BUILD OS:", buildOS)
```
Comment (oseoin) [inline cmd/nginx-ingress/main.go:68]:
```suggestion
```

Comment (AlexFenlon) [reply cmd/nginx-ingress/main.go]: Thanks, just noticed that, it was a debug.

**Review by shaun-nx — APPROVED** [review]

**Review by ADubhlaoich — COMMENTED** [review]
LGTM, though I am not approving since this is primarily a code PR.

**Review by pdabelf5 — APPROVED** [review]

**Review by jjngx — APPROVED** [review]

**Pull request merged**